viewing_party/party.py

```diff
@@ -1,23 +1,217 @@
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    """
+    input: user_data dictionary with a "watched" list of movie dictionaries
+    output: average rating float
+    """
+
+    average_rating = 0.0
+    movie_ratings = [] 
+
+    for watched in user_data["watched"]:
+        rating = watched["rating"]
+        movie_ratings.append(rating)
+    
+    sum_of_movie_ratings = sum(movie_ratings)
+    length_of_movie_ratings = len(movie_ratings)
+    if length_of_movie_ratings == 0:
+        return average_rating
+    average_rating = sum_of_movie_ratings/length_of_movie_ratings
+
+    return average_rating
+
+
+def get_most_watched_genre(user_data):
+    """
+    input: user_data dictionary with a "watched" list of movie dictionaries
+    output: genre string
+    """
+
+    most_frequently_watched_genre = None
+    movie_genres = []
+
+    for watched in user_data["watched"]:
+        genre = watched["genre"]
+        movie_genres.append(genre)
+    if len(movie_genres) == 0:
+        return None
+
+    most_frequently_watched_genre = (mode(movie_genres))
+    return most_frequently_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    """
+    input: user_data dictionary with a "watched" list of movie dictionaries
+    output: list of dictionaries (which represents list of movies)
+    """
+
+    user_watched_list = []
+    friends_watched_list = []
+    movies_user_watched_that_friends_didnt_watch = []
+    
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_list.append(movie)
+
+    for user in user_data["watched"]:
+        user_watched_list.append(user)
+
+
+    for movie in user_watched_list: 
+        if movie not in friends_watched_list:
+            movies_user_watched_that_friends_didnt_watch.append(movie)
+            
+
+    return movies_user_watched_that_friends_didnt_watch
+
+
+def get_friends_unique_watched(user_data):
+    """
+    input: user_data dictionary with a "watched" list of movie dictionaries
+    output: list of dictionaries (which represents list of movies)
+    """
+    user_watched_list = []
+    friends_watched_list = []
+    movies_friends_watched_that_user_didnt_watch = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched_list.append(movie)
+
+    for user in user_data["watched"]:
+        user_watched_list.append(user)
+
+    for movie in friends_watched_list:
+        if movie not in user_watched_list and movie not in movies_friends_watched_that_user_didnt_watch:
+            movies_friends_watched_that_user_didnt_watch.append(movie)
+
+    return movies_friends_watched_that_user_didnt_watch

```

Comment:
Similar comments from `get_unique_watched` can be applied to this function (using list comprehension and helper function for the friends' movies list).